drivers/rawexec/driver_test.go

```diff
@@ -662,3 +679,255 @@ func TestRawExecDriver_validate(t *testing.T) {
 		})
 	}
 }
+
+func TestRawExecDriver_buildEnvList(t *testing.T) {
+	defaultEnvironment := genEnv()
+
+	testCases := []struct {
+		name             string
+		taskConfig       *TaskConfig
+		driverTaskConfig *drivers.TaskConfig
+		driverConfig     *Config
+		deniedPresent    bool
+	}{
+		{name: "OK, no globs",
+			taskConfig: &TaskConfig{
+				DeniedEnvvars: []string{"AWS_SECRET_KEY"},
+			},
+			driverTaskConfig: &drivers.TaskConfig{
+				Env: defaultEnvironment,
+			},
+			driverConfig: &Config{
+				DeniedEnvvars: []string{"NOMAD_TOKEN", "GITHUB_TOKEN", "GITHUB_PASSWORD"},
+			},
+			deniedPresent: false,
+		},
+		{name: "OK, globs",
+			taskConfig: &TaskConfig{
+				DeniedEnvvars: []string{"AWS_SECRET_KEY"},
+			},
+			driverTaskConfig: &drivers.TaskConfig{
+				Env: defaultEnvironment,
+			},
+			driverConfig: &Config{
+				DeniedEnvvars: []string{"*_TOKEN"},
+			},
+			deniedPresent: false,
+		}, {name: "OK, only one set",
+			taskConfig: &TaskConfig{
+				DeniedEnvvars: []string{},
+			},
+			driverTaskConfig: &drivers.TaskConfig{
+				Env: defaultEnvironment,
+			},
+			driverConfig: &Config{
+				DeniedEnvvars: []string{"*_TOKEN"},
+			},
+			deniedPresent: false,
+		},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			present := false
+			foundVars := []string{}
+
+			d := newEnabledRawExecDriver(t)
+			d.config = tc.driverConfig
+			envList := d.buildEnvList(tc.taskConfig, tc.driverTaskConfig)
+			deniedList := strings.Join(slices.Concat(tc.taskConfig.DeniedEnvvars, tc.driverConfig.DeniedEnvvars), ", ")
+
+			for _, v := range envList {
+				key, _, found := strings.Cut(v, "=")
+				if !found {
+					t.Fatalf("Unable to parse environment variable list, cannot evaluate")
+				}
+				keyGlob := "*" + key + "*"
+				if glob.Glob(keyGlob, deniedList) {
+					present = true
+					foundVars = append(foundVars, key)
+				}
+			}
+
+			if present != tc.deniedPresent {
+				t.Fatalf("Expected deniedPresent to be %t, got %t, list= %#v", tc.deniedPresent, present, foundVars)
+			}
+		})
+	}
+}
+
+func TestRawExecDriver_Env(t *testing.T) {
+	ci.Parallel(t)
+
+	require := require.New(t)
```

Comment:
Nomad uses https://pkg.go.dev/github.com/shoenig/test over testify, although you'll find there's still quite a few older tests using it that we've never got around to converting. We try to convert older tests as we touch them. `shoenig/test` uses generics to fix some meaningful equality checking gotchas in testify, and improves ergonomics over testify in a non-backwards-compatible way (ex. making sure the expectation is in a consistent order in the arguments list).